### PR TITLE
Form validation as function + tests.

### PR DESCRIPTION
**Pull request by sl33kr**
Pull request for #572. Still have to update type defs.

### CODE REVIEW

**Comment by davidkpiano** [comment]
Code looks good to me! I can make the type def update if you want, unless you're already on it.

**Comment by sl33kr** [comment]
I haven't done it yet no sorry.

There were a couple of lines in there I had comments on as I wasn't sure the best way to handle them.

File: src/components/form-component.js [hunk 1]
```diff
@@ -197,12 +196,64 @@ function createFormClass(s = defaultStrategy) {
...
+
+        const currentValue = field
+          ? s.get(modelValue, field)
+          : modelValue;
+
+        // If the validators didn't change, the validity didn't change.
+        if ((!initial && !validatorsChanged) && (nextValue === currentValue)) {
+          // TODO this will only set the errors on form when using the function.
+          // How handle? Safe to assume will be no dispatch?
+          // fieldsErrors[field] = getField(formValue, field).errors;
```
Comment (sl33kr) [inline src/components/form-component.js]: Do you have an idea on how to correctly handle this? Is the assumption ok?

**Comment by sl33kr** [comment]
I'll do type def now.

File: src/components/form-component.js (hunk 1 above)
Comment (davidkpiano) [inline src/components/form-component.js]: Yes, this is a safe assumption - do nothing if the values are the same after initial load.

**Comment by davidkpiano** [comment]
Thank you! I'll revise the comments.

**Pull request merged**